### PR TITLE
fix: ui improvements

### PR DESCRIPTION
**Pull request by uhliksk**
## Description

- Fix pagination is showing more pages than actually used and is not compliant with standard pagination methods.
- Add support to show up to all 12 columns on ultra wide monitors.
- Unexecuted grid trades compacted in lists with more than 10 items.
- Object for pagination consistent across the application.
- Fix unnecessary text wrapping if there is enough space to print a text.
- Show Action time in local time.
- Changed time format for logs to 24 hours to be consistent within application.

## Related Issue

#437, #440, #442, #444, #446

## Motivation and Context

The fix will improve the user experience.

## How Has This Been Tested?

Tested on UHD display and on wide multi monitor setup with tens of grid trades in various states.

## Screenshots (if appropriate):

Buttons are visible for available pages only:

image

Ultra wide monitor support:

image

Compact grid trade list:

image

Pagination object at top and bottom with up to 8 numbered buttons:

image

Even a very wide texts are no longer wrapped after the fix is applied:

image

Columns are consistent after the fix is applied:

image

### CODE REVIEW

**Comment by chrisleekr** [comment]
@uhliksk 
Thanks for merging PRs. Amazing contributions!

I will review+test after #448 is merged.
I am very lack of time, so please bear with me. :)  

Thanks!

**Comment by uhliksk** [comment]
@chrisleekr I understand. I made some minor changes in the meanwhile to even improve this PR.

**Comment by chrisleekr** [comment]
@uhliksk 

This is a very good fix. 
I also was very annoyed by pagination, but your fix made me comfortable. haha.
And CSS fix. 👍

File: public/js/CoinWrapperBuySignal.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ class CoinWrapperBuySignal extends React.Component {
     } = symbolConfiguration;
 
     const buyGridRows = gridTrade.map((grid, i) => {
-      return (
+      const modifiedGridTradeIndex = Math.min( Math.max ( currentGridTradeIndex , 7 ) , gridTrade.length - 4 );
```
Comment (chrisleekr) [inline public/js/CoinWrapperBuySignal.js:44]: I am not sure about this change.

Are you saying you have a grid settings of more than 10? Wow...
I am not sure there are traders who are using that many grid settings.

Ok, let's say you do have lots of grid settings, the below screenshot made me so confused "Where is my Grid 11?"

Screen Shot 2022-08-02 at 22 08 44

Is it possible you can display a collapsed indicator such as `...5 grid trade hidden...` for hidden grid settings?
And when click the column, it shows the hidden columns. 

I just don't want to hide without any indicator.

I know I am asking for more changes (So sorry 🙏). If you don't have time, let's keep it original.
Again, sorry to ask for changes. @uhliksk

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js:44]: I'm actually using 200 grid trades. I wanted the change in the code as simple as possible but I can add the indicator.

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js:44]: Screenshot I put in PR section is the real thing. I just wanted to save resources not rendering the whole thing even when collapsed.

image

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js:44]: Is this OK?

image

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js]: @chrisleekr Did you reviewed the last screenshot?

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js]: Grids with two indicators are taller than grids with single indicator. I'll fix it today to keep grid height consistent.

image

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js]: Fixed. There are always 10 rows visible including indicators to keep the grid trade height uniform.

image

Comment (chrisleekr) [reply public/js/CoinWrapperBuySignal.js]: @uhliksk 

Looks good. :+1: Thanks for the update.

Good to merge?

Comment (uhliksk) [reply public/js/CoinWrapperBuySignal.js]: @chrisleekr Yes, I think everything is good to go now. Thank you.

**Comment by chrisleekr** [comment]
@uhliksk 

It's merged!

Thanks for your contributions!

**Pull request merged**